beacon/goclient/goclient.go

```diff
@@ -147,7 +147,7 @@ type goClient struct {
 }
 
 // New init new client and go-client instance
-func New(logger *zap.Logger, opt beaconprotocol.Options, operatorID spectypes.OperatorID, slotTicker slot_ticker.Ticker) (beaconprotocol.BeaconNode, error) {
+func New(logger *zap.Logger, opt beaconprotocol.Options, operatorID spectypes.OperatorID, slotTickerProvider slot_ticker.SlotTickerFactory) (beaconprotocol.BeaconNode, error) {
```

Comment:
What do you think about `slot_ticker.SlotTickerProvider` for consistency with the variable name? That would also get rid of the Java-like `Factory` suffix which is usually not used in Go